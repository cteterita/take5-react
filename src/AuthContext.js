import React from 'react';

const AuthContext = React.createContext({
  currentUser: null,
  signInWithGoogle: null,
  signOut: null,
});

export default AuthContext;
