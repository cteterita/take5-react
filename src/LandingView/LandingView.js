import React, { useContext } from 'react';

import AuthContext from '../AuthContext';

function LandingView() {
  const auth = useContext(AuthContext);
  return (
    <section id="daily-entry">
      <h3>Take 5 minutes each day for yourself</h3>
      <p>Set your intentions in the morning</p>
      <p>Practice gratitude in the evening</p>
      <button type="submit" onClick={auth.signInWithGoogle}>
        Sign in with Google
      </button>
      <button type="submit" onClick={auth.signInAnonymously}>
        Try it Anonymously
      </button>
    </section>
  );
}

export default LandingView;
