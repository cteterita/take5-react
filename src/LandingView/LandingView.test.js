import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import LandingView from './LandingView';
import AuthContext from '../AuthContext';

const testAuth = {
  signInWithGoogle: () => {},
  signInAnonymously: () => {},
};

describe('LandingView component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <AuthContext.Provider value={testAuth}>
        <LandingView />
      </AuthContext.Provider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders as expected', () => {
    const tree = renderer.create(
      <AuthContext.Provider value={testAuth}>
        <LandingView />
      </AuthContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
