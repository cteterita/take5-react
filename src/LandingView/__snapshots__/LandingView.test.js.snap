// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingView component renders as expected 1`] = `
<section
  id="daily-entry"
>
  <h3>
    Take 5 minutes each day for yourself
  </h3>
  <p>
    Take the first 2.5 minutes after waking up to set your intentions for the day.
  </p>
  <p>
    Return in the evening for 2.5 minutes to review your day and practice gratitude.
  </p>
  <div
    id="sign-in-holder"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Sign in with Google
    </button>
     or 
    <button
      onClick={[Function]}
      type="submit"
    >
      Try it Anonymously
    </button>
  </div>
</section>
`;
